### PR TITLE
Split single attribute into two separate ones and modifications to subject/type reflection code

### PR DESCRIPTION
**Pull request by sirkirby**
- two new attributes replacing the former sink attribute. One for type and one for subject, which makes for cleaner usage
- refactored reflection naming for type and subject based on feedback
- resolved issues with parameter naming including extra characters
- updated readme to reflect changes
- updated nuget config in csproj to point to this repo and not the default serilog repo. copyright was also updated

### CODE REVIEW

**Comment by sirkirby** [comment]
proposing as the v1.0.0 release to nuget

File: src/Serilog.Sinks.EventGrid/Sinks/EventGrid/EventGridSink.cs [hunk 1]
```diff
@@ -99,41 +99,67 @@ void GetEventInfoFromAttribute(CustomEventRequest customEvent)
...
+      {
+        // then look for the first class with the attribute, there can be only one
+        var classTypeAttribute = methods.FirstOrDefault(m => m.ReflectedType != null && m.ReflectedType.GetCustomAttribute<EventGridTypeAttribute>() != null)?.ReflectedType?.GetCustomAttribute<EventGridTypeAttribute>();
+        typeAttribute = classTypeAttribute;
       }
+      return typeAttribute;
+    }
 
-      string GetType()
+    private EventGridSubjectAttribute GetSubjectFromAttribute(MethodBase[] methods)
```
Comment (jiaxiangl) [inline src/Serilog.Sinks.EventGrid/Sinks/EventGrid/EventGridSink.cs]: looks like the subject function and type function are kind of same. could we merge this two into one function and pass T as generic type? something like   
private T GetFromAttribute<T>(MethodBase[] methods)

Comment (sirkirby) [reply src/Serilog.Sinks.EventGrid/Sinks/EventGrid/EventGridSink.cs]: @jiaxiangl absolutely! good idea. I'll make the change

**Review by jiaxiangl — APPROVED** [review]
Awesome! others looks good to me.

**Comment by sirkirby** [comment]
@jiaxiangl hows that? had to introduce a new interface and refactor the attributes a bit, but now down to one generic reflection method in the sink... so the type of string retrieved is simply based on the type of attribute we are looking for, which was the desired usage in the client code. 

```csharp
// both types are just IEventGridAttributes with a property of CustomValue
[EventGridSubject("string"), EventGridType("string)]]
public void MyMethod(){}
```

**Pull request merged**